internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -962,3 +973,64 @@ func (md *machineDeployment) doSmokeChecks(ctx context.Context, lm machine.Leasa
 
 	return fmt.Errorf("smoke checks for %s failed: %v", lm.Machine().ID, err)
 }
+
+func (md *machineDeployment) checkDNS(ctx context.Context) error {
+	client := client.FromContext(ctx).API()
+	ipAddrs, err := client.GetIPAddresses(ctx, md.appConfig.AppName)
+	if err != nil {
+		return err
+	}
+
+	if url := md.appConfig.URL(); url != nil && len(ipAddrs) > 0 {
+		iostreams := iostreams.FromContext(ctx)
+		fmt.Fprintf(iostreams.ErrOut, "Checking DNS configuration for %s\n", md.colorize.Bold(url.Host))
+
+		fqdn := dns.Fqdn(url.Host)
+		c := dns.Client{
+			Dialer:       &net.Dialer{Timeout: time.Minute},
+			Timeout:      time.Minute,
+			DialTimeout:  time.Minute,
+			ReadTimeout:  time.Minute,
+			WriteTimeout: time.Minute,
+		}
+
+		b := backoff.NewExponentialBackOff()
+		b.InitialInterval = 1 * time.Second
+		b.MaxElapsedTime = 10 * time.Second
+
+		return backoff.Retry(func() error {
+			m := new(dns.Msg)
+
+			var numIPv4, numIPv6 int
+			for _, ipAddr := range ipAddrs {
+				ip := net.ParseIP(ipAddr.Address)
+				if ip.To4() != nil {
+					numIPv4 += 1
+				} else {
+					numIPv6 += 1
+				}
+			}
+
+			m.SetQuestion(fqdn, dns.TypeA)
+			answerv4, _, err := c.Exchange(m, "8.8.8.8:53")
```

Comment:
It looks like we use `9.9.9.9` in `flyctl doctor`. It might be better to use a consistent DNS resolver? https://github.com/superfly/flyctl/blob/7cb1642ee5931c99292c69ad67eaa7fa3b1dbfda/internal/command/doctor/app_checks.go#L199